airflow/www_rbac/filters.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+from flask import g
+from flask_appbuilder.models.sqla.filters import BaseFilter
+
+from airflow.www_rbac.app import appbuilder
+
+
+class AirflowFilter(BaseFilter):
+    """
+    Add utility function to make BaseFilter easy and fast
+
+    These utility function exist in the SecurityManager, but would do
+    a database round trip at every check. Here we cache the role objects
```

Comment:
Is there really caching taking place here? Looking for caching logic and didn't find any. From my understanding `g.user.role` is local at this point and has all the details whether we use it or not so security assertion are pretty cheap pretty cheap.